src/batou_ext/fcio.py

```diff
@@ -305,6 +309,69 @@ def get_diff(self, old, new):
         return result
 
 
+class DirectoryXMLRPC(batou.component.Component):
+    def configure(self):
+        self.xmlrpc = create_xmlrpc_client(self.environment)
+        self.provide("directory-xmlrpc", self.xmlrpc)
+
+
+class MaintenanceStart(batou.component.Component):
+    def configure(self):
+        self.require("needs-maintenance", strict=False, reverse=True)
+        self.xmlrpc = self.require_one("directory-xmlrpc")
+
+    def verify(self):
+        raise batou.UpdateNeeded()
+
+    def update(self):
+        change_maintenance_state(
+            self.xmlrpc, self.host.name, desired_state=True
+        )
+
+
+class MaintenanceEnd(batou.component.Component):
+    def configure(self):
+        self.require("needs-maintenance", strict=False)
+        self.xmlrpc = self.require_one("directory-xmlrpc")
+
+    def verify(self):
+        raise batou.UpdateNeeded()
+
+    def update(self):
+        change_maintenance_state(
+            self.xmlrpc, self.host.name, desired_state=False
+        )
+
+
+def change_maintenance_state(
+    xmlrpc, host_name, desired_state, predict_only=False
+):
+    rg_name = host_name[:-2]
```

Comment:
I effectively assume here that each host in a deployment is in the same RG.
Do we even have cases where this is not True?

Reply:
Can be modified now.